WIP: MDEV-20544: Update bundled Groonga and Mroonga

otegami opened the pull request:
This is still in work in progress. I opened this PR just for using CI. I'll update the PR description after this is completed.
- [x] *The Jira issue number for this PR is: MDEV-20544*

<!--
An amazing description should answer some questions like:
1. What problem is the patch trying to solve?
2. If some output changed that is not visible in a test case, what was it looking like before the change and how it's looking with this patch applied?
3. Do you think this patch might introduce side-effects in other parts of the server?
-->
## Description
TODO: fill description here

## Release Notes
TODO: What should the release notes say about this change?
Include any changed system variables, status variables or behaviour. Optionally list any https://mariadb.com/kb/ pages that need changing.

## How can this PR be tested?

TODO: modify the automated test suite to verify that the PR causes MariaDB to behave as intended.
Consult the documentation on ["Writing good test cases"](https://mariadb.org/get-involved/getting-started-for-developers/writing-good-test-cases-mariadb-server).
<!--
In many cases, this will be as simple as modifying one `.test` and one `.result` file in the `mysql-test/` subdirectory.
Without automated tests, future regressions in the expected behavior can't be automatically detected and verified.
-->

If the changes are not amenable to automated testing, please explain why not and carefully describe how to test manually.

<!--
Tick one of the following boxes [x] to help us understand if the base branch for the PR is correct.
-->
## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature or a refactoring, and the PR is based against the `main` branch.*
- [ ] *This is a bug fix, and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

<!--
  All code merged into the MariaDB codebase must meet a quality standard and coding style.
  Maintainers are happy to point out inconsistencies but in order to speed up the review and merge process we ask you to check the CODING standards.
-->
## PR quality check
- [ ] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/-/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [ ] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

kou committed:
Update bundled Groonga and Mroonga

otegami commented:
Thanks for the pointers! We’ll tackle these one by one.

> appears to be some dependencies - can you provide a list.

In the meantime, here are the packages I believe we’ll need:

## Groonga (Ubuntu Build-Depends)

- libedit-dev
- libevent-dev
- liblz4-dev
- libmecab-dev
- libmsgpack-dev
- libsimdjson-dev
- libssl-dev
- libstemmer-dev
- libthrift-dev
- libxxhash-dev
- libzmq3-dev
- libzstd-dev
- rapidjson-dev
- zlib1g-dev

ref: https://github.com/groonga/groonga/blob/7b50b0cd0ca6815fb430a63e945e7220693dd3d7/packages/debian/control#L9-L26
ref: https://github.com/groonga/groonga/blob/main/packages/apt/debian-bookworm/Dockerfile#L24-L47

## Mroonga (MariaDB 11.4)

- groonga-devel
- groonga-normalizer-mysql-devel
- MariaDB-client(RPM)
- MariaDB-common(RPM)
- MariaDB-server(RPM)
- MariaDB-shared(RPM)

ref: https://github.com/mroonga/mroonga/blob/main/packages/mariadb-11.4-mroonga/yum/mariadb-11.4-mroonga.spec.in#L32-L37

otegami committed:
fix Debian packaging for updated Mroonga/Groonga bundles

After updating bundled Groonga and Mroonga, the Debian
CI build was failing with dh_missing errors.

The upstream CMake files now install additional files
that weren't accounted for in the Debian packaging.

- Update debian/mariadb-plugin-mroonga.install to
  include new files
- Add distribution-specific scripts to
  debian/not-installed to exclude them from packaging
  since Debian has its own packaging mechanisms

This aligns the Debian packaging with the upstream
installation targets while maintaining proper Debian
packaging practices.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MariaDB/server?pullRequest=4185) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/MariaDB/server?pullRequest=4185) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: kou<br/>:x: otegami<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MariaDB/server?pullRequest=4185) it.</sub>

abetomo committed:
Fix `dh_missing: warning`

Fix the following errors:

```
h_missing --fail-missing
dh_missing: warning: usr/share/doc/groonga-normalizer-mysql/groonga-normalizer-mysql/README.md exists in debian/tmp but is not installed to anywhere (related file: "debian/tmp/usr/share/groonga/README.md")
dh_missing: warning: usr/share/doc/groonga-normalizer-mysql/groonga-normalizer-mysql/lgpl-2.0.txt exists in debian/tmp but is not installed to anywhere (related file: "debian/tmp/usr/share/groonga-normalizer-mysql/lgpl-2.0.txt")
dh_missing: warning: usr/share/groonga/onigmo/AUTHORS exists in debian/tmp but is not installed to anywhere (related file: "debian/tmp/usr/share/mariadb/mroonga/AUTHORS")
dh_missing: warning: usr/share/groonga/onigmo/COPYING exists in debian/tmp but is not installed to anywhere (related file: "debian/tmp/usr/share/groonga/COPYING")
dh_missing: warning: usr/share/groonga/onigmo/HISTORY exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/groonga/onigmo/README exists in debian/tmp but is not installed to anywhere (related file: "debian/tmp/usr/share/mariadb/policy/selinux/README")
dh_missing: warning: usr/share/groonga/rapidjson/CHANGELOG.md exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/groonga/rapidjson/license.txt exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/groonga/rapidjson/readme.md exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/mariadb/mroonga/deb/postinst.sh exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/mariadb/mroonga/deb/prerm.sh exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/mariadb/mroonga/rpm/post.sh exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/mariadb/mroonga/rpm/preun.sh exists in debian/tmp but is not installed to anywhere
dh_missing: warning: usr/share/mariadb/mroonga/update.sql exists in debian/tmp but is not installed to anywhere
dh_missing: error: missing files, aborting
	While detecting missing files, dh_missing noted some files with a similar name to those
	that were missing.  This error /might/ be resolved by replacing references to the
	missing files with the similarly named ones that dh_missing found - assuming the content
	is identical.
	As an example, you might want to replace:
	 * debian/tmp/usr/share/groonga/README.md
	with:
	 * usr/share/doc/groonga-normalizer-mysql/groonga-normalizer-mysql/README.md
	in a file in debian/ or as argument to one of the dh_* tools called from debian/rules.
	(Note it is possible the paths are not used verbatim but instead directories
	containing or globs matching them are used instead)
	Alternatively, add the missing file to debian/not-installed if it cannot and should not
	be used.
	The following debhelper tools have reported what they installed (with files per package)
	 * dh_install: libmariadb-dev (154), libmariadb-dev-compat (4), libmariadb3 (6), libmariadb3-compat (2), libmariadbclient18 (1), libmariadbd-dev (5), libmariadbd19 (1), libmysqlclient18 (1), mariadb-backup (6), mariadb-client (51), mariadb-client-compat (38), mariadb-client-core (6), mariadb-common (1), mariadb-plugin-columnstore (121), mariadb-plugin-connect (2), mariadb-plugin-connect-jdbc (1), mariadb-plugin-cracklib-password-check (2), mariadb-plugin-gssapi-client (1), mariadb-plugin-gssapi-server (2), mariadb-plugin-hashicorp-key-management (3), mariadb-plugin-mroonga (5), mariadb-plugin-oqgraph (2), mariadb-plugin-provider-bzip2 (2), mariadb-plugin-provider-lz4 (2), mariadb-plugin-provider-lzma (2), mariadb-plugin-provider-lzo (2), mariadb-plugin-provider-snappy (2), mariadb-plugin-rocksdb (8), mariadb-plugin-s3 (4), mariadb-plugin-spider (2), mariadb-server (83), mariadb-server-compat (12), mariadb-server-core (47), mariadb-test (54), mariadb-test-data (6), mysql-common (0)
	If the missing files are installed by another tool, please file a bug against it.
	When filing the report, if the tool is not part of debhelper itself, please reference the
	"Logging helpers and dh_missing" section from the "PROGRAMMING" guide for debhelper (10.6.3+).
	  (in the debhelper package: /usr/share/doc/debhelper/PROGRAMMING.gz)
	Be sure to test with dpkg-buildpackage -A/-B as the results may vary when only a subset is built
	If the omission is intentional or no other helper can take care of this consider adding the
	paths to debian/not-installed.
dh_install: error: dh_missing --fail-missing returned exit code 25
make: *** [debian/rules:200: binary] Error 25
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
```

otegami commented:
@grooverdan 

About https://github.com/mroonga/mroonga/issues/943#issuecomment-3055161056

Thanks again for the guidance. The Mroonga team has been working hard on this,
but unfortunately we’re unlikely to finish everything by July 24 due to two major blockers at the moment.
We’ll keep pushing on and will update you as soon as we have progress.

1. Windows CI (buildbot/amd64-windows)

The Windows builder is failing with platform‑specific errors.
We’re also treating all warnings (e.g. C4244) as errors, and there are too many to resolve before the release date.
For example, ii.cpp in Groonga alone generated enormous errors, and we estimate around a thousand such locations across the codebase.

ref: https://buildbot.mariadb.org/#/builders/234/builds/40629
ref: https://github.com/abetomo/groonga/actions/runs/16311415283/job/46067698311#step:13:2714

2. Compile‑only CI (buildbot/amd64-compile-only-nopart-debug)

We’re encountering ISO C90 "mixed declarations and code" errors in dependencies like H3 and llama.cpp's ggml.
H3's CMakeLists forces C99 features (see https://github.com/uber/h3/blob/2257ace292aeaa1d21ac06b157ca916c012f5add/CMakeLists.txt#L362),
and llama.cpp errors appear as follows.

```
error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
    const uint32_t bits = fp32_to_bits(base);
```

While disabling these modules would clear these errors because they aren't really necessary for Mroonga functions, there are likely many more hidden issues in other modules, and we currently do not have enough time to fix them all by July 24.

ref: https://buildbot.mariadb.org/#/builders/904/builds/111

kou commented:
> 1. Windows CI (buildbot/amd64-windows)

We can disable some warning compile options in the first step. We can work on suppressing warnings later.

> 2. Compile‑only CI (buildbot/amd64-compile-only-nopart-debug)

We can disable the `declaration-after-statement` warning by adding `-Wno-declaration-after-statement`.

kou commented:
No.

MariaDB adds `-Wdeclaration-after-statement`: https://github.com/MariaDB/server/blob/ce7ab467dbdfb5fc83a22bd53d923f7d0d1c6a3e/cmake/maintainer.cmake#L25

It's related.

(I think that MariaDB can remove the warning.)

ParadoxV5 commented:
> (I think that MariaDB can remove the warning.)

Maybe MariaDB had it there to enforce [a coding style](https://github.com/MariaDB/server/blob/main/CODING_STANDARDS.md#variable-declarations)?

abetomo committed:
add storage/mroonga submodule

abetomo committed:
update submodules

abetomo committed:
update to the latest mroonga

abetomo committed:
update to the latest groonga
